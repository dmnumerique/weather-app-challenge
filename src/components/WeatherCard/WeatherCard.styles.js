import styled from 'styled-components'
import theme from '../../theme/theme.styles'

export const Card = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0 ${theme.spacing.s2} 0 0;

  @media ${theme.device.tablet} {
    padding: 0 ${theme.spacing.s2} 0 0;
  }
`

export const Day = styled.p`
  font-weight: bold;
  font-size: ${theme.fonts.size.p};
  margin: 0 0 ${theme.spacing.s1} 0;

  @media ${theme.device.tablet} {
    font-size: ${theme.fonts.size.h5};
  }
`

export const Icon = styled.img`
  width: 60%;
  margin: 0 0 ${theme.spacing.s1} 0;

  @media ${theme.device.tablet} {
    width: 75%;
  }
`

export const Temp = styled.span`
  color: ${(props) => (props.isHigh ? theme.color.black : theme.color.gray)};
  font-weight: ${(props) =>
    props.isHigh ? theme.fonts.weight.semi : theme.fonts.weight.regular};
  margin-right: ${(props) => (props.isHigh ? theme.spacing.s1 : 0)};
  font-size: ${theme.fonts.size.sm};

  @media ${theme.device.tablet} {
    font-size: ${theme.fonts.size.p};
  }
`
