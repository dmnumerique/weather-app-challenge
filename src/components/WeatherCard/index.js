import React from 'react'
import PropTypes from 'prop-types'
import { Card, Day, Icon, Temp } from './WeatherCard.styles'
import { iconsPath } from '../../consts/config'

const WeatherCard = ({ date, icon, description, min, max }) => {
  const getDayName = () => {
    const days = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat']
    const today = new Date().getDay()
    const d = new Date(date).getDay()
    return d === today ? 'Today' : days[d]
  }

  return (
    <Card>
      <Day>{getDayName()}</Day>
      <Icon src={`${iconsPath}${icon}.png`} alt={description} />
      <div>
        <p>
          <Temp isHigh>{Math.round(max)}&deg;</Temp>
          <Temp>{Math.round(min)}&deg;</Temp>
        </p>
      </div>
    </Card>
  )
}

WeatherCard.propTypes = {
  date: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  min: PropTypes.number.isRequired,
  max: PropTypes.number.isRequired
}

export default WeatherCard
