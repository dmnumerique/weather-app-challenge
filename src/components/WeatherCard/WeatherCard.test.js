/* eslint-disable react/jsx-props-no-spreading */
import renderer from 'react-test-renderer'
import React from 'react'
import WeatherCard from './index'

const defaultProps = {
  date: '2021-03-17',
  icon: 'c01d',
  description: 'Windy',
  min: 17,
  max: 22
}

it('should render with default props', () => {
  const tree = renderer.create(<WeatherCard {...defaultProps} />).toJSON()
  expect(tree).toMatchSnapshot()
})
