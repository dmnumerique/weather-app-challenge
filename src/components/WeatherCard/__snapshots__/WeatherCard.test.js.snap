// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with default props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px 0 0;
}

.c1 {
  font-weight: bold;
  font-size: 16px;
  margin: 0 0 8px 0;
}

.c2 {
  width: 60%;
  margin: 0 0 8px 0;
}

.c3 {
  color: #000000;
  font-weight: 600;
  margin-right: 8px;
  font-size: 14px;
}

.c4 {
  color: #727272;
  font-weight: 400;
  margin-right: 0;
  font-size: 14px;
}

@media (min-width:768px) {
  .c0 {
    padding: 0 16px 0 0;
  }
}

@media (min-width:768px) {
  .c1 {
    font-size: 18px;
  }
}

@media (min-width:768px) {
  .c2 {
    width: 75%;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 16px;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 16px;
  }
}

<div
  className="c0"
>
  <p
    className="c1"
  >
    Today
  </p>
  <img
    alt="Windy"
    className="c2"
    src="assets/icons/c01d.png"
  />
  <div>
    <p>
      <span
        className="c3"
      >
        22
        °
      </span>
      <span
        className="c4"
      >
        17
        °
      </span>
    </p>
  </div>
</div>
`;
